testing/run-scripts/run_cloud.sh

```diff
@@ -75,89 +78,95 @@ fi
 #  - pull from DNS
 if [ -z "$PUBLIC_IP" ]
 then
-  if docker ps -a | grep uproxy-sshd >/dev/null
-  then
-    if [ `docker inspect --format='{{ .State.Status }}' uproxy-sshd` != "running" ]
+    if docker ps -a | grep uproxy-sshd >/dev/null
     then
-      docker start uproxy-sshd > /dev/null
+        if [ `docker inspect --format='{{ .State.Status }}' uproxy-sshd` != "running" ]
+        then
+            docker start uproxy-sshd > /dev/null
+        fi
+        # Don't fail if the current installation has no /hostname.
+        PUBLIC_IP=`docker exec uproxy-sshd cat /hostname || echo -n ""`
     fi
-    # Don't fail if the current installation has no /hostname.
-    PUBLIC_IP=`docker exec uproxy-sshd cat /hostname || echo -n ""`
-  fi
-  if [ -z "$PUBLIC_IP" ]
-  then
-    # Beautiful cross-platform one-liner cogged from:
-    #   http://unix.stackexchange.com/questions/22615/how-can-i-get-my-external-ip-address-in-bash
-    PUBLIC_IP=`dig +short myip.opendns.com @resolver1.opendns.com`
-  fi
+    if [ -z "$PUBLIC_IP" ]
+    then
+        # Beautiful cross-platform one-liner cogged from:
+        #   http://unix.stackexchange.com/questions/22615/how-can-i-get-my-external-ip-address-in-bash
+        PUBLIC_IP=`dig +short myip.opendns.com @resolver1.opendns.com`
+    fi
+fi
+
+if [ $UPDATE = true ]
```

Comment:
Let's mirror how the other fields, e.g. banner, start the container if it's been stopped. This might work:
```bash
if [ -z "$INVITE_CODE" ]
then
  if docker ps -a | grep uproxy-sshd >/dev/null
  then
    if [ `docker inspect --format='{{ .State.Status }}' uproxy-sshd` != "running" ]
    then
      docker start uproxy-sshd > /dev/null
    fi
    INVITE_CODE=`docker cp uproxy-sshd:/initial-giver-invite-code -|tar xO`
  fi
fi
```

(and we can factor out a function to start a container in a followup)